tools/iceoryx_build_test.sh

```diff
@@ -196,36 +196,27 @@ cd $BUILD_DIR
 echo " [i] Current working directory: $(pwd)"
 
 echo ">>>>>> Start building iceoryx package <<<<<<"
-cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DBUILD_STRICT=$STRICT_FLAG -DCMAKE_INSTALL_PREFIX=$ICEORYX_INSTALL_PREFIX -DCMAKE_EXPORT_COMPILE_COMMANDS=$QACPP_JSON -DTOML_CONFIG=on -Dtest=$TEST_FLAG -Dcoverage=$COV_FLAG -Droudi_environment=on -Dexamples=OFF -Dintrospection=$INTROSPECTION_FLAG -Ddds_gateway=$DDS_GATEWAY_FLAG -Dbinding_c=ON -DONE_TO_MANY_ONLY=$ONE_TO_MANY_ONLY_FLAG -Dsanitize=$SANITIZE_FLAG $WORKSPACE/iceoryx_meta
+cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DBUILD_STRICT=$STRICT_FLAG -DCMAKE_INSTALL_PREFIX=$ICEORYX_INSTALL_PREFIX -DCMAKE_EXPORT_COMPILE_COMMANDS=$QACPP_JSON -DTOML_CONFIG=on -Dtest=$TEST_FLAG -Dcoverage=$COV_FLAG -Droudi_environment=on -Dexamples=ON -Dintrospection=$INTROSPECTION_FLAG -Ddds_gateway=$DDS_GATEWAY_FLAG -Dbinding_c=ON -DONE_TO_MANY_ONLY=$ONE_TO_MANY_ONLY_FLAG -Dsanitize=$SANITIZE_FLAG $WORKSPACE/iceoryx_meta
 cmake --build . --target install -- -j$NUM_JOBS
 echo ">>>>>> Finished building iceoryx package <<<<<<"
 
 # Dont build examples when coverage or sanitization is enabled
-if [ "$COV_FLAG" == "ON" ] || [ "$SANITIZE_FLAG" == "ON" ]
-then
-    echo ">>>>>> Skip building iceoryx examples <<<<<<"
+if [ "$SANITIZE_FLAG" == "ON" ]; then
+    echo ">>>>>> skip Out-of-tree build <<<<<<"
+elif [ "$COV_FLAG" == "ON" ]; then
+    echo ">>>>>> initial gcov scan <<<<<<"
+    $WORKSPACE/tools/gcov/lcov_generate.sh $WORKSPACE initial #make an initial scan to cover also files with no coverage
 else
-    echo ">>>>>> Start building iceoryx examples <<<<<<"
+    echo ">>>>>> Start building Out-of-tree build<<<<<<"
     cd $BUILD_DIR
-    mkdir -p iceoryx_examples
+    mkdir -p out_of_tree_build
     echo ">>>>>>>> icedelivery"
     cd $BUILD_DIR/iceoryx_examples
     mkdir -p icedelivery
     cd icedelivery
     cmake -DCMAKE_PREFIX_PATH=$ICEORYX_INSTALL_PREFIX -DCMAKE_INSTALL_PREFIX=$ICEORYX_INSTALL_PREFIX $WORKSPACE/iceoryx_examples/icedelivery
     cmake --build . --target install -- -j$NUM_JOBS
-    echo ">>>>>>>> iceperf"
-    cd $BUILD_DIR/iceoryx_examples
-    mkdir -p iceperf
```

Comment:
is iceperf not build anymore?

Reply:
oh, now I see it. The examples are build with iceoryx_meta. I'd suggest to revert that, and build all of them out of tree. It's a good test to check a breakage before they reach our users

Reply:
Is it necessary to build all the examples out-of-tree when you only want to check if the cmake package is correctly generated? The icedelivery example does a find_package iceoryx_posh which then resolves all underlying dependencies.
In this case we are using the installed packages in ../install/prefix. 
If i know that the package is correctly found in one example, why should i repeat it then with other examples?

C_Binding should be added to the out-of-tree build thats right.

Reply:
does the icedelivery example use all of our API? If not, how do you ensure that all headers are installed and maybe some are not?

Reply:
i reverted the changes in the script and will do it as part of https://github.com/eclipse/iceoryx/issues/433.
A cleaner solution would be to have the out-of-tree build as part of iceoryx_meta instead of having it in the bash script. I guess i can do that with `ExternalProject_Add`. For this PR this is out of scope.